plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/QueryInsightsService.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.lifecycle.AbstractLifecycleComponent;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightsExporter;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightsExporterType;
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+import org.opensearch.plugin.insights.settings.QueryInsightsSettings;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+/**
```

Comment:
I'm in favour of keeping a single service for QueryInsights, which should accept - 

1. List of metricType it supports - like topNLatency, topNCpu, topNHeap etc.
2. List of exporters - local_index/system_index, file, logs, Otel etc.

It should provide ability to - 
1. register/unregister or enable/disable exporter on the fly (controlled by dynamic cluster setting).
2. enable/disable metricType on the fly (again controlled by dynamic cluster setting).
3. start() and stop() will start all enabled metricType(). stop() will do the same.

Reply:
Hi @rishabhmaurya ! Thanks for the suggestion. I refactored the code based on your input.

Reply:
thanks @ansjcy for incorporating them. I have added few comments.

Reply:
@rishabhmaurya  -- are you okay with the change to use `QueryInsightsService` for ingestion of each query record, while `TopQueriesService` handles the processing for each `MetricType`?

Reply:
looks good to me, thanks.